components/director/cmd/tenantfetcher-svc/main.go

```diff
@@ -158,29 +166,74 @@ func main() {
 
-	transact, closeFunc, err := persistence.Configure(ctx, cfg.Handler.Database)
-	exitOnError(err, "Error while establishing the connection to the database")
-	defer func() {
-		err := closeFunc()
-		exitOnError(err, "error while closing the connection to the database")
+	systemFieldDiscoveryProcessor := systemfielddiscoveryengine.NewOperationProcessor(systemFieldDiscoverySvc)
+	onDemandChannel := make(chan string, 100)
+
+	handler := initAPIHandler(ctx, httpClient, cfg, tenantSynchronizers, saasRegistryOperationsManager, onDemandChannel)
+	runMainSrv, shutdownMainSrv := createServer(ctx, cfg, handler, "main")
+
+	go func() {
+		<-ctx.Done()
+		// Interrupt signal received - shut down the servers
+		shutdownMainSrv()
 	}()
 
-	webhookProcessor := webhookprocessor.NewWebhookProcessor(transact, webhookSvc, tenantSvc, appSvc, webhookClient, cfg.WebhookProcessorElectionConfig, cfg.WebhookProcessorJobInterval, cfg.SystemFieldDiscoveryWebhookPartialProcessing, cfg.SystemFieldDiscoveryWebhookPartialProcessMaxDays)
+	for i := 0; i < cfg.ParallelOperationProcessors; i++ {
+		go func(ctx context.Context, opManager *operationsmanager.OperationsManager, opProcessor *systemfielddiscoveryengine.OperationsProcessor, executorIndex int) {
+			for {
+				select {
+				case <-onDemandChannel:
+				default:
+				}
+
+				processedOperationID, err := claimAndProcessOperation(ctx, opManager, opProcessor)
+				if err != nil {
+					log.C(ctx).Errorf("Failed during claim and process operation %q by executor %d . Err: %v", processedOperationID, executorIndex, err)
+				}
+				if len(processedOperationID) > 0 {
+					log.C(ctx).Infof("Processed Operation: %s by executor %d", processedOperationID, executorIndex)
+				} else {
+					// Queue is empty - no operation claimed
+					log.C(ctx).Infof("No Processed Operation by executor %d", executorIndex)
+
+					select {
+					case operationID := <-onDemandChannel:
+						log.C(ctx).Infof("Operation %q send for processing through OnDemand channel to executor %d", operationID, executorIndex)
+					case <-time.After(cfg.OperationProcessorQuietPeriod):
+						log.C(ctx).Infof("Quiet period finished for executor %d", executorIndex)
+					}
+				}
+			}
+		}(ctx, saasRegistryOperationsManager, systemFieldDiscoveryProcessor, i)
+	}
+
 	go func() {
-		if err := webhookProcessor.StartWebhookProcessorJob(ctx, webhookprocessor.SaaSRegistryLabelValue); err != nil {
-			log.C(ctx).WithError(err).Error("Failed to run WebhookProcessorJob. Stopping app...")
+		if err := saasRegistryOperationsManager.StartRescheduleOperationsJob(ctx, []string{model.OperationStatusCompleted.ToString()}); err != nil {
+			log.C(ctx).WithError(err).Error("Failed to run  RescheduleOperationsJob. Stopping app...")
+			cancel()
+		}
+	}()
+
+	go func() {
+		if err := saasRegistryOperationsManager.StartDeleteOperationsJob(ctx); err != nil {
+			log.C(ctx).WithError(err).Error("Failed to run  RescheduleOperationsJob. Stopping app...")
```

Comment:
`RescheduleOperationsJob` -> `DeleteOperationsJob`
There is also double spacebar between `run` and `RescheduleOperationsJob`